### PR TITLE
Finish updating internal API in libcrmcommon

### PR DESCRIPTION
**Pull request by kgaillot**
This PR finishes the "best practices" project for internal API in libcrmcommon, to follow current naming/copyright conventions, return new standard return codes where appropriate, and add more doxygen comment blocks.

These particular changes are mostly renames and splitting one file into two.

### CODE REVIEW

**Comment by kgaillot** [comment]
@clumens , can you review this? No rush, it's low priority.

File: daemons/attrd/attrd_commands.c [hunk 1]
```diff
@@ -617,7 +618,8 @@ attrd_peer_message(crm_node_t *peer, xmlNode *xml)
             clear_attribute_value_seen();
         }
 
-        for (child = __xml_first_child(xml); child != NULL; child = __xml_next(child)) {
+        for (child = pcmk__xml_first_child(xml); child != NULL;
+             child = pcmk__xml_next(child)) {
```
Comment (clumens) [inline daemons/attrd/attrd_commands.c:622]: We use this pattern so much.  It might be worth a separate PR that introduces something like `pcmk__xml_child_foreach` that takes a parent and a callback function, and the callback function type looks like you'd expect.

**Comment by clumens** [comment]
This all looks good to me.

File: daemons/attrd/attrd_commands.c (hunk 1 above)
Comment (kgaillot) [reply daemons/attrd/attrd_commands.c:622]: It would be cleaner, but I'd be concerned about the function call overhead (these are static inline functions currently) since XML processing happens a whole lot. Some timing comparisons might help decide.

Comment (clumens) [reply daemons/attrd/attrd_commands.c:622]: I think it'd be worth looking into.  If you have any good pointers on how to do useful timing comparisons, I'd be happy to look at it... eventually.

Comment (kgaillot) [reply daemons/attrd/attrd_commands.c:622]: Taking the largest cts/scheduler/*.xml file and running crm_simulate with a big --repeat would probably be a decent approach. Or creating a test CIB with 1000 nodes and 1000 resources and a short test program with nested loops through those. It's probably insignificant, but I'd want to be sure.

We have several related functions:
* pcmk__xml_first_child()/pcmk__xml_next() return XML nodes of any kind other than text (document, element, comment, CDATA, ...)
* pcmk__xe_first_child()/pcmk__xe_next() return just XML elements
* first_named_child()/crm_next_same_xml() return just XML elements of a given type

Currently we don't always use the proper pair for a given context, so that should be part of any overhaul. E.g. we almost certainly have places where it would choke on a comment because it's using the first pair when it should be using one of the others. And we have lots of places that could use the last pair but are string-comparing the element type themselves (for no good reason, they were probably just written before those existed).
